modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2259 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Navier-Stokes includes
+#include "NSFVAction.h"
+#include "NS.h"
+
+// MOOSE includes
+#include "FEProblem.h"
+#include "MooseObject.h"
+#include "NonlinearSystemBase.h"
+#include "RelationshipManager.h"
+
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_variables");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_user_objects");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_ics");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_kernels");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_bcs");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_material");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_pps");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_materials");
+
+InputParameters
+NSFVAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("This class allows us to set up Navier-Stokes equations for porous "
+                             "medium or clean fluid flows using incompressible or weakly "
+                             "compressible approximations with a finite volume discretization.");
+
+  /**
+   * General parameters used to set up the simulation.
+   */
+  params.addParam<std::vector<SubdomainName>>(
+      "block", "The list of blocks on which NS equations are defined on");
+
+  MooseEnum sim_type("steady-state transient", "steady-state");
+  params.addParam<MooseEnum>("simulation_type", sim_type, "Navier-Stokes equation type");
+
+  MooseEnum comp_type("incompressible weakly-compressible", "incompressible");
+  params.addParam<MooseEnum>(
+      "compressibility", comp_type, "Compressibility constraint for the Navier-Stokes equations.");
+
+  params.addParam<bool>(
+      "porous_medium_treatment", false, "Whether to use porous medium kernels or not.");
+
+  MooseEnum turbulence_type("mixing-length none", "none");
+  params.addParam<MooseEnum>(
+      "turbulence_handling",
+      turbulence_type,
+      "The way additional diffusivities are determined in the turbulent regime.");
```

Comment:
Yes, you end up creating a nu_t with almost every turbulence model.

Reply:
At the end it would be great, but we need to generalize our turbulence Reynolds-stress kernels first to accept a general nu_t material. (I think this will also open the door to use the Cauchy stress tensor instead of FV diffusion. At the moment we have a MixingLength kernel which expect mixing length. I'll put this on the list, but I would like to do that in a separate PR.